spec/tasks/data_indexer_spec.rb

```diff
@@ -0,0 +1,208 @@
+      @di.send(:define_doc)
+      # @di.send(:index_mods)
+    end
+    it 'is a pending test'
+    # it 'should have received an add' do
+    #   #TODO: How do I test solr received an add
+    #   expect(stub_solr).to receive(:add).at_least(1).times.and_return(true)
+    #   expect(stub_solr).to receive(:add).at_least(1).times.and_return(arg) |do|
+    #     expect(arg).to have_key 'responseHeader'
+    #     expect(arg['responseHeader']).to have_key 'status'
+    #     expect(arg['responseHeader']['status']).to eq 0
+    #   end
+    # end
+  end
+
+  describe '#index_annotations' do
+    before do
+      @di = DataIndexer.new('test collection', nil, 'http://dms-data.stanford.edu/data/manifests/Stanford/kq131cs7229/stub/manifest.json')
+      @di.send(:fetch_manifest)
+      @di.send(:define_doc)
+      # @di.send(:index_mods)
+      # @ans = @di.send(:index_annotations)
+    end
+    it 'is a pending test'
+    # TODO: How do I test solr received 44 adds (1  + 19 + 1 + 23)
+    # it 'should have received 44 adds' do
+    #   expect(stub_solr).to receive(:add).at_least(44).times.and_return(true)
+    # end
+    # it 'should have counted 2 lists do'
+    #   expect(@ans[0]).to eq(2)
+    # end
+    # it 'should have counted 44 resources do'
+    #   expect(@ans[1]).to eq(44)
+    #   expect(@ans[2]).to eq(44)
+    # end
+  end
+
+  describe '#index' do
+    before do
+      @di = DataIndexer.new('test collection', nil, 'http://dms-data.stanford.edu/data/manifests/Stanford/kq131cs7229/stub/manifest.json')
+      # @di.index
+    end
+    it 'is a pending test'
+    # TODO: How do I test solr received 45 adds (1 + 1  + 19 + 1 + 23)
+    # it 'should have received 45 adds' do
+    #   expect(stub_solr).to receive(:add).at_least(45).times.and_return(true)
+    # end
+  end
+
+  describe '#index_csv' do
+    before do
+      @di = DataIndexer.new('test collection', manifest_csv_file, nil)
+      # @di.index_csv
+    end
+    it 'is a pending test'
+    # TODO: How do I test solr received lots of adds
+    # it 'should have received adds' do
+    #   expect(stub_solr).to receive(:add).at_least(50).times.and_return(true)
+    # end
+  end
+
+  describe '#run index' do
+    before do
+      @di = DataIndexer.new('test collection', nil, 'http://dms-data.stanford.edu/data/manifests/Stanford/kq131cs7229/stub/manifest.json')
+      @di.run
+    end
+    it 'is a pending test'
+    # it 'should have received adds' do
+    #   expect(stub_solr).to receive(:add).at_least(45).times.and_return(true)
+    #   expect(stub_solr).to receive(:commit).once_and_return(true)
+    # end
+  end
+
+  describe '#run index_csv' do
+    before do
+      @di = DataIndexer.new('test collection', manifest_csv_file, nil)
+      @di.run
+    end
+    it 'is a pending test'
+    # it 'should have received adds' do
+    #   expect(stub_solr).to receive(:add).at_least(50).times.and_return(true)
+    #   expect(stub_solr).to receive(:commit).once_and_return(true)
+    # end
```

Comment:
@anusharanganathan, I'm not sure this will work w/ all of your tests here, but a quick experiment pulling this down locally I was able to get this test to pass by making a few minor changes.

1. `@di.run` out of the `before` block and at the end of the it block.
2. Changed `stub_solr` to `@stub_solr`
3. Change `.once_and_return` to `.once.and_return`

```ruby
describe '#run index_csv' do
  before do
    @di = DataIndexer.new('test collection', manifest_csv_file, nil)
  end
  it 'should have received adds' do
    expect(@stub_solr).to receive(:add).at_least(50).times.and_return(true)
    expect(@stub_solr).to receive(:commit).once.and_return(true)
    @di.run
  end
end
```

Some of the other tests may need other changes, but hopefully that gets you moving forward.